src/xtgeo/grid3d/grid.py

```diff
@@ -2151,12 +2276,18 @@ def activate_all(self):
 
         self._tmp = {}
 
-    def inactivate_by_dz(self, threshold):
+    def inactivate_by_dz(self, threshold: float) -> None:
         """Inactivate cells thinner than a given threshold."""
         _grid_etc1.inactivate_by_dz(self, threshold)
         self._tmp = {}
 
-    def inactivate_inside(self, poly, layer_range=None, inside=True, force_close=False):
+    def inactivate_inside(
+        self,
+        poly: xtgeo.Polygons,
```

Comment:
Consider pulling just `Polygons` in through the `TYPE_CHECKING` guard too, just for consistency with xtgeo class types (e.g. Points)

Reply:
We could, but since we already do an `import xtgeo` i think the impact would be zero in this case?